plugins/modules/ec2_vpc_vgw_info.py

```diff
@@ -150,20 +145,22 @@ def get_virtual_gateway_info(virtual_gateway):
 
 def list_virtual_gateways(client, module):
     params = dict()
+    vpn_gateway_ids = module.params.get("vpn_gateway_ids")
+    filters = module.params.get("filters")
 
-    params["Filters"] = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
-
-    if module.params.get("vpn_gateway_ids"):
-        params["VpnGatewayIds"] = module.params.get("vpn_gateway_ids")
+    if filters:
+        params["Filters"] = ansible_dict_to_boto3_filter_list(filters)
+    if vpn_gateway_ids:
+        params["VpnGatewayIds"] = vpn_gateway_ids
 
     try:
-        all_virtual_gateways = client.describe_vpn_gateways(**params)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to list gateways")
+        all_virtual_gateways = describe_vpn_gateways(client, **params)
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
```

Comment:
error handling is done here - https://github.com/ansible-collections/amazon.aws/pull/2331/files#diff-11d6a927433f45aa84de2f54fa0adf33e8094ebb417c75ea476ffe61b4e4f587R652

Reply:
removed.